### PR TITLE
port scala xml to cats effect 3

### LINKED ISSUE #3841: Port scala-xml to cats-effect-3

### PR DESCRIPTION
**Pull request by m-sp**
fixes #3841

### CODE REVIEW

**Review by hamnis — APPROVED** [review]

File: scala-xml/src/main/scala/scalaxml/ElemInstances.scala [hunk 1]
```diff
@@ -44,7 +44,7 @@ trait ElemInstances {
     *
     * @return an XML element
     */
-  implicit def xml[F[_]](implicit F: Sync[F]): EntityDecoder[F, Elem] = {
+  implicit def xml[F[_]](implicit F: Async[F]): EntityDecoder[F, Elem] = {
```
Comment (rossabaker) [inline scala-xml/src/main/scala/scalaxml/ElemInstances.scala:47]: Does anything prevent this being `Concurrent`? That's the new constraint for most of the decoders.  Is it the `F.delay` on line 56?

**Review by rossabaker — COMMENTED** [review]
Thanks! One question, out of my ignorance on cats-effect-3.

File: scala-xml/src/main/scala/scalaxml/ElemInstances.scala (hunk 1 above)
Comment (m-sp) [reply scala-xml/src/main/scala/scalaxml/ElemInstances.scala:47]: Exactly, I tried concurrent first but delay isn't available on concurrent

Comment (rossabaker) [reply scala-xml/src/main/scala/scalaxml/ElemInstances.scala:47]: I don't see much of a side effect here.  It consumes the `source`, but it's our source, from a strict byte array.  We don't need to demand a tighter constraint for this local mutation.  I think we could replace that `F.delay` with a `try / catch`, sort out the errors, and lose the `Async` constraint.

Comment (m-sp) [reply scala-xml/src/main/scala/scalaxml/ElemInstances.scala]: thanks for the review, makes sense. looks also a bit simpler

**Comment by m-sp** [comment]
seems like the test failure is unrelated?

**Review by rossabaker — APPROVED** [review]
Yes, rerunning the tests. :+1: on green.

**Pull request merged**